src/awkward/_broadcasting.py

```diff
@@ -308,28 +298,23 @@ def one_to_one_parameters_factory(
     `inputs`. If the two values match, then a list of parameter objects is returned,
     where each element of the returned list corresponds to the parameters of the
     content at the same position in the `inputs` sequence. If the length of the
-    given contents does not match the requested list length, a ValueError is raised.
+    given contents does not match the requested list length, the intersection of the parameters
+    is returned instead.
     """
-    # Find the parameters of the inputs, with None values for non-Contents
-    input_parameters = [_parameters_of(c, default=None) for c in inputs]
-
-    def apply(n_outputs) -> list[dict[str, Any] | None]:
-        if n_outputs != len(inputs):
-            raise ValueError(
-                "cannot follow one-to-one parameter broadcasting rule for actions "
-                "which change the number of outputs."
-            )
-        return input_parameters
-
-    return apply
+    if n_outputs == len(parameters):
+        return [p if p is not NO_PARAMETERS else None for p in parameters]
+    else:
+        return intersection_parameters_factory(parameters, n_outputs)
```

Comment:
New default behaviour: perform the intersection when we don't have the same number of inputs as outputs